third_party/terraform/tests/resource_bigquery_table_test.go

```diff
@@ -503,6 +528,55 @@ resource "google_bigquery_table" "test" {
 `, bucketName, datasetID, tableID)
 }
 
+func testAccBigQueryTableHivePartitioningCustomSchema(bucketName, datasetID, tableID string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "test" {
+	name          = "%s"
+	force_destroy = true
+}
+
+resource "google_storage_bucket_object" "test" {
+	name    = "key1=20200330/data.json"
+        content = "{\"name\":\"test\", \"last_modification\":\"2020-04-01\"}"
+	bucket  = google_storage_bucket.test.name
+}
+
+resource "google_bigquery_dataset" "test" {
+        dataset_id = "%s"
+}
+
+resource "google_bigquery_table" "test" {
+	table_id   = "%s"
+	dataset_id = google_bigquery_dataset.test.dataset_id
+
+	external_data_configuration {
+            source_format = "NEWLINE_DELIMITED_JSON"
+            autodetect = false
+            source_uris= ["gs://${google_storage_bucket.test.name}/*"]
+
+            hive_partitioning_options {
+                mode = "CUSTOM"
+                source_uri_prefix = "gs://${google_storage_bucket.test.name}/{key1:STRING}"
+	    }
+
+            schema = <<EOH
+[
+	{
+		"name": "name",
+		"type": "STRING"
+	},
+	{
+		"name": "last_modification",
+		"type": "DATE"
+	}
+]
+EOH
+        }
+	depends_on = ["google_storage_bucket_object.test"]
```

Comment:
this depends_on shouldn't be necessary, since that resource is already interpolated from this one (line 559)

Reply:
this is to make sure the bucket object exists and not the bucket itself.